integration/spark-common/src/main/scala/org/apache/spark/util/FileUtils.scala

```diff
@@ -91,4 +94,24 @@ object FileUtils {
     }
   }
 
+  /*
+   * The store location of carbon comes from three places:
+   * 1. default location path in code(../carbon.store)
+   * 2. configurate "carbon.storelocation" in carbon.properties
+   * 3. pass the location as the parameter
+   * The priority is low to high.
+   */
+  def getHighestPriorityStorePath(storePathParameter: String): String = {
+    val storePathDefaultValue =
+      new File(CarbonCommonConstants.STORE_LOCATION_DEFAULT_VAL).getCanonicalPath
+    val storePathProperties = CarbonProperties.getInstance()
+      .getProperty(CarbonCommonConstants.STORE_LOCATION)
+    if (!storePathParameter.equals(storePathDefaultValue)) {
```

Comment:
@jackylk My first idea is like yours, but I worry about it will change the API. If it can be changed, I will do this solution.